### PR TITLE
`factory()`

### PR DESCRIPTION
**Pull request by kyscott18**

### CODE REVIEW

File: examples/feature-factory/ponder.config.ts [hunk 1]
```diff
@@ -20,21 +20,21 @@ export default createConfig({
     LlamaCore: {
       network: "sepolia",
       abi: LlamaCoreAbi,
-      factory: {
+      address: factory({
```
Comment (0xOlias) [inline examples/feature-factory/ponder.config.ts:23]: Looks so good!

File: packages/core/src/build/configAndIndexingFunctions.ts [hunk 2]
```diff
@@ -504,20 +505,13 @@ export async function buildConfigAndIndexingFunctions({
...
       const resolvedAddress = source?.address;
 
-      if (resolvedFactory !== undefined && resolvedAddress !== undefined) {
-        throw new Error(
-          `Validation failed: Contract '${contractMetadata.name}' cannot specify both 'factory' and 'address' options.`,
-        );
-      }
-
-      if (resolvedFactory) {
+      if (typeof resolvedAddress === "object") {
```
Comment (0xOlias) [inline packages/core/src/build/configAndIndexingFunctions.ts:510]: Are we validating the factory arguments at runtime anywhere?

File: packages/core/src/config/config.ts [hunk 3]
```diff
@@ -200,54 +180,36 @@ type ContractsConfig<networks, contracts> = {} extends contracts
...
-            [name in allNetworkNames]?: Prettify<
-              GetAddress<unknown> & TransactionReceiptConfig & BlockConfig
-            >;
-          };
-    };
+> = {
+  /**
+   * Network that this account is deployed to. Must match a network name in `networks`.
+   * Any filter information overrides the values in the higher level "accounts" property.
+   * Factories cannot override an address and vice versa.
```
Comment (0xOlias) [inline packages/core/src/config/config.ts:191]: Is this comment now irrelevant?

**Review by 0xOlias — APPROVED** [review]
Looks great! So much clearer this way. Couple nits.

File: packages/core/src/build/configAndIndexingFunctions.ts (hunk 2 above)
Comment (kyscott18) [reply packages/core/src/build/configAndIndexingFunctions.ts:510]: https://github.com/ponder-sh/ponder/blob/kjs/factory/packages/core/src/build/factory.ts

File: packages/core/src/config/config.ts (hunk 3 above)
Comment (kyscott18) [reply packages/core/src/config/config.ts]: nice catch

**Review by 0xOlias — APPROVED** [review]
LGTM

**Pull request merged**